src/gallium/drivers/tegra/tegra_stream.c

```diff
@@ -104,21 +104,16 @@ static int tegra_allocate_cmdbuf(struct drm_tegra *drm,
  * tegra_stream_push().
  */
 
-int tegra_stream_create(struct drm_tegra *drm, struct tegra_stream *stream,
+int tegra_stream_create(struct drm_tegra *drm,
+                        struct drm_tegra_channel *channel,
+                        struct tegra_stream *stream,
                         uint32_t words_num)
 {
-    int ret;
-
-    ret = drm_tegra_channel_open(&stream->channel, drm, DRM_TEGRA_GR2D);
-    if (ret) {
-        fprintf(stderr, "drm_tegra_channel_open() failed %d\n", ret);
-        goto err_open_channel;
-    }
-
     words_num = words_num ? words_num : BUFFER_SIZE_WORDS;
 
     stream->status      = TEGRADRM_STREAM_FREE;
     stream->buffer_size = words_num;
+    stream->channel     = channel;
 
```

Comment:
@digetx: If you want to have a loo, here you go: https://github.com/kusma/mesa/commit/883c11860de157125302b4f40118ffedd9c16493

However, we can't just do this on top of there without introducing a leak, as the current version (on my branch) transfers the owership of the channel, but there's no corresponding cleanup-code added yet at the higher level.